app/javascript/src/components/Dashboard/Notes/index.jsx

```diff
@@ -1,118 +1,110 @@
 import React, { useState } from "react";
 
 import EmptyNotesListImage from "images/EmptyNotesList";
-import { Delete } from "neetoicons";
-import { Button, PageLoader, Toastr } from "neetoui";
-import { Container, Header, SubHeader } from "neetoui/layouts";
+import { Button, Toastr } from "neetoui";
+import { Container, Header } from "neetoui/layouts";
 import { useTranslation } from "react-i18next";
 
-import notesApi from "apis/notes";
 import DeleteAlert from "components/commons/DeleteAlert";
 import EmptyState from "components/commons/EmptyState";
+import MenuBar from "components/commons/MenuBar";
 
-import { TABLE_ROW_DATA } from "./constants";
+import Card from "./Card";
+import { MENU_BAR_OPTIONS, TABLE_ROW_DATA } from "./constants";
 import NewNotePane from "./Pane/Create";
-import Table from "./Table";
+import EditNotePane from "./Pane/Edit";
 
 const Notes = () => {
-  const [loading, setLoading] = useState(false);
   const [showNewNotePane, setShowNewNotePane] = useState(false);
   const [showDeleteAlert, setShowDeleteAlert] = useState(false);
+  const [showEditNote, setShowEditNote] = useState(false);
+  const [selectedNote, setSelectedNote] = useState({});
   const [searchTerm, setSearchTerm] = useState("");
-  const [selectedNoteIds, setSelectedNoteIds] = useState([]);
+  const [selectedNoteId, setSelectedNoteId] = useState(null);
   const [notes, setNotes] = useState(TABLE_ROW_DATA);
+  const [showMenu, setShowMenu] = useState(true);
 
   const { t } = useTranslation();
 
-  const fetchNotes = async () => {
-    try {
-      setLoading(true);
-      const {
-        data: { notes },
-      } = await notesApi.fetch();
-      setNotes(notes);
-    } catch (error) {
-      logger.error(error);
-    } finally {
-      setLoading(false);
-    }
-  };
-
   const deleteNotes = () => {
-    const nonDeletedNotes = notes.filter(
-      note => !selectedNoteIds.includes(note.id)
-    );
+    const nonDeletedNotes = notes.filter(note => note.id !== selectedNoteId);
     setNotes(nonDeletedNotes);
-    Toastr.success(
-      t("note.delete.successWithCount", { count: selectedNoteIds.length })
-    );
+    Toastr.success(t("note.delete.success"));
   };
 
-  if (loading) {
-    return <PageLoader />;
-  }
+  const handleDeleteNote = noteId => {
+    setSelectedNoteId(noteId);
+    setShowDeleteAlert(true);
+  };
+
+  const handleEditNote = noteId => {
+    const selectedEditNote = notes.find(note => note.id === noteId) || {};
```

Comment:
I didn't understand the need for the OR part. What's the default value of the state?

Reply:
I added it to avoid a type error as shown in the image below.
<img width="873" alt="Screenshot 2023-02-09 at 7 01 02 PM" src="https://user-images.githubusercontent.com/35962711/217826268-2ea9a492-5249-4915-bc8c-8cace8cbfb95.png">

Reply:
Please don't go after fixing type errors in VSCode. The source of truth for errors should always be the terminal. VSCode red squiggly lines can be ignored.